recipe/meta.yaml

```
@@ -93,7 +93,7 @@ outputs:
         - python
       run:
         - python
-        - pyqt >=5.10  # [not osx and not aarch64 and not ppc64le and python_impl == "cpython"]
+        - pyside6 >=6.7.2=*_1  # [not osx and not aarch64 and not ppc64le and python_impl == "cpython"]
```

Since we're changing this dependency, can we re-validate the selectors?

Currently this is equivalent to "cpython on linux-64 & win-64" only, yet we have [builds](https://anaconda.org/conda-forge/pyside6/files) for pyside6 for every architecture bar ppc64le (though to be fair, pyqt seems to have full architecture support, so I don't know where those constraints came from).

However, in the best case, this could be reduced to
```suggestion
        - pyside6 >=6.7.2=*_1  # [python_impl == "cpython" and not ppc64le]
```

---

Also, we should get rid of the `=*_1` part. I think this is going to stick beyond the version, so that for _every_ `pyside6` version, we'd only allow build 1. The higher build number will already be pulled in by default, and this is actually guaranteed because https://github.com/conda-forge/pyside2-feedstock/pull/230 changed nothing about the requirements of the package (and if `_0` and `_1` are exactly equivalent from the POV of the solver, `_1` always wins).

---

I assume the `not osx` comes from the fact that the default GUI backend for matplotlib on macOS is the `macOSX` backend, not the Qt backend. But maybe it does make sense to install the Qt libraries anyway for consistency? That way anyone who installs `matplotlib` will be guaranteed that the Qt backend is available. Right now they wouldn't be guaranteed that.

---

> I assume the not osx comes from the fact that the default GUI backend for matplotlib on macOS is the macOSX backend, not the Qt backend. But

That is the source of the `not osx`. I think the main arguement for keeping the `not osx` is to save bandwidth to mac users who may not ever use the qt dependency. I'm not persuaded by the consistency argument, if the user wants pyqt it for some other reason they should explicitly install it and if another package needs qt that package should depend on it.  Currently mac users are not getting qt installed implicitly via Matplotlib so keeping that state should not break anyone.